extras/ams_wrapper/src/api/models/model.py

```diff
@@ -28,43 +28,34 @@
 from src.api.ovms_connector import OvmsUnavailableError, ModelNotFoundError, OvmsConnector
-from src.api.models.input_config import ModelInputConfiguration, \
-    ModelInputConfigurationSchema, ValidationError
+from src.api.models.model_config import ModelInputConfiguration, \
+    ModelInputConfigurationSchema, ValidationError, ModelOutputConfiguration, \
+    ModelOutputConfigurationSchema
 
 logger = get_logger(__name__)
 
 
 class Model(ABC):
     def __init__(self, model_name: str, ovms_connector: OvmsConnector,
-                 labels_path: str, config_file_path: str = None):
+                 config_file_path: str = None):
         self.ovms_connector = ovms_connector
         self.model_name = model_name
-        self.labels = self.load_labels(labels_path)
-        self.input_configs: Dict[str, ModelInputConfiguration] = None # load input configuration
-        if config_file_path:
-            self.input_configs = self.load_input_configs(config_file_path)
-
-    def load_labels(self, labels_path):
-        try:                                                                          
-            with open(labels_path, 'r') as labels_file:
-                data = json.load(labels_file)
-                labels = dict()
-                for output in data['outputs']: 
-                    labels[output["output_name"]] = output['classes']
-        except Exception as e:                                                        
-            logger.exception("Error occurred while opening labels file: {}".format(e))
-            sys.exit(1)
-        return labels
+        #self.labels = self.load_labels(config_file_path)
+        self.input_configs = self.load_input_configs(config_file_path)
+        self.output_configs = self.load_output_configs(config_file_path) 
+        self.labels = {output_name: {index: label for label, index in self.output_configs[output_name].classes.items()}
+                       for output_name in self.output_configs.keys()}
 
     def preprocess_binary_image(self, binary_image: bytes) -> np.ndarray:
-        # By default the only performed preprocessing is converting image
-        # from binary format to numpy ndarray. If a model requires more specific
-        # preprocessing this method should be implemented in its class.
-        try:
-            preprocessed_image = default_preprocessing(binary_image)
+        try: 
+            # Assuming single input for now
+            preprocessing_config = next(iter(self.input_configs.values()))
+            preprocessing_config = preprocessing_config.as_preprocessing_options()
+            preprocessed_image = default_preprocessing(binary_image, **preprocessing_config)
             preprocessed_image = np.expand_dims(preprocessed_image, axis=0)
         except Exception as e:
             # TODO: Error handling
+            logger.exception('Failed to preprocess binary image')
```

Comment:
Lets add e message.

Reply:
I think it's the part conflicting with target branch. 
This exception here shall be re-raised.

Reply:
Exception is now re-raised, regarding `e` message - it is automatically included when using `logger.exception()`.